docs/training/apis-howto/api-pytorch-trainer-ug.rst

```diff
@@ -0,0 +1,354 @@
+             validation_period=pytorch.Batch(10),
+       )
+
+Local + Distributed Training
+============================
+
+Local training can utilize multiple GPUs on a single node with a few modifications to the above
+code.
+
+.. note::
+
+   Only Horovod and PyTorch Distributed backends are supported.
+
+.. code::
+
+    def main():
+   +     # Initialize distributed backend before pytorch.init()
+   +     dist.init_process_group(backend="gloo|nccl")
+
+   +     # Set flag used by internal PyTorch training loop
+   +     os.environ["USE_TORCH_DISTRIBUTED"] = "true"
+
+   +     # Initialize DistributedContext specifying chief IP
+         with det.pytorch.init(
+   +       distributed=core.DistributedContext.from_torch_distributed  (chief_ip="localhost")
+         ) as train_context:
+             trial = MNistTrial(train_context)
+             trainer = det.pytorch.Trainer(trial, train_context)
+             trainer.fit(
+                 max_length=pytorch.Epoch(1),
+                 checkpoint_period=pytorch.Batch(10),
+                 validation_period=pytorch.Batch(10),
+             )
+
+Call your distributed backend's launcher directly: `torchrun --nproc_per_node=4 train.py`.
+
+Local Training - Test Mode
+==========================
+
+PyTorch Trainer accepts a test_mode parameter which, if true, trains and validates your training
+code for only one batch, then exits. This is helpful for debugging code.
+
+.. code::
+
+    trainer.fit(
+                 max_length=pytorch.Epoch(1),
+                 checkpoint_period=pytorch.Batch(10),
+                 validation_period=pytorch.Batch(10),
+   +             # Train and validate 1 batch, then exit.
+   +             test_mode=True
+             )
+
+This is the same codepath as
+
+.. code:: bash
+
+   det e create det.yaml . --local --test
+
+**************************************************************************
+ Step 4: Prepare Your Training Code for Deploying to a Determined Cluster
+**************************************************************************
+
+Once you are satisfied with the results of training the model locally, you submit the code to a
+cluster.
+
+**Example workflow of frequent iterations between local debugging and cluster deployment**
+
+This code should allow for local and cluster training with no code changes.
+
+.. code::
+
+    def main():
+   +   local = det.get_cluster_info() is None
+   +   if local:
+   +       # (Optional) Initialize distributed backend before pytorch.init()
+   +       dist.init_process_group(backend="gloo|nccl")
+   +       # Set flag used by internal PyTorch training loop
+   +       os.environ["USE_TORCH_DISTRIBUTED"] = "true"
+   +       distributed_context = core.DistributedContext.from_torch_distributed  (chief_ip="localhost")
+   +       # (Optional) Pass in an exp conf and instance of hparams if training code needs it
+   +       expconf = yaml.safe_load(pathlib.Path("./config.yaml"))
+   +       hparams = {"lr": 0.02}
```

Comment:
Let's not document the legacy compatibility code path as our recommended code path.